File: admin/src/dtos/CityName.ts
import Common from "../utility/Common";
import { CityRes } from "./City";
import { LanguageRes } from "./Language";
import { PagedReq } from "./PagedReq";

export interface CityNameRes {
  cityNameId?: string;
  name?: string;
  
  languageId?: string;
  language?: LanguageRes;
  
  cityId?: string;
  city?: CityRes;
}

export class CityNameReqEdit {
  languageId?: string = "";
  cityId?: string = "";
  name?: string = "";

  constructor(languageId?: string, cityId?: string) {
    this.languageId = languageId;
    this.cityId = cityId;
  }
}

export class CityNameReqSearch extends PagedReq {
  cityId?: string;
  constructor(
    {
      pageNumber = 1,
      pageSize = Common.DEFAULT_PAGE_SIZE,
      orderBy = "",
      searchText = "",
    }: PagedReq,
    {cityId = ""}
  ) {
    super({
      pageNumber: pageNumber,
      pageSize: pageSize,
      orderBy: orderBy,
      searchText: searchText,
    });
    this.cityId = cityId;
  }
}

File: dto/dtos/CityName.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using dto.Paging;

namespace dto.dtos
{
    public class CityNameRes
    {
        public int CityNameId { get; set; }

        public int? LanguageId { get; set; }
        public LanguageRes? Language { get; set; }

        public int? CityId { get; set; }
        public CityRes? City { get; set; }

        public string? Name { get; set; }
    }

    public class CityNameReqEdit
    {
        [Required]
        public int? LanguageId { get; set; }
        
        [Required]
        public int? CityId { get; set; }
        
        [Required]
        public string? Name { get; set; }
    }

    public class CityNameReqSearch : PagedReq
    {
        [Required]
        public int? CityId { get; set; }
    }
}
